server/src/test/java/com/paypal/selion/grid/servlets/GridForceRestartDelegateServletTest.java

```diff
@@ -0,0 +1,86 @@
+|  with the License.                                                                                                  |
+|                                                                                                                     |
+|  You may obtain a copy of the License at                                                                            |
+|                                                                                                                     |
+|       http://www.apache.org/licenses/LICENSE-2.0                                                                    |
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+package com.paypal.selion.grid.servlets;
+
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Tests for {@link GridForceRestartDelegateServlet}
+ */
+// TODO Add tests which consider SeLionRemoteProxy instances
+public class GridForceRestartDelegateServletTest extends BaseGridRegistyServletTest {
+    private static GridForceRestartDelegateServlet servlet;
+
+    @BeforeClass
+    public void beforeClass() {
+        super.beforeClass();
+        // Initialize the servlet under test
+        servlet = new GridForceRestartDelegateServlet(registry);
+    }
+
+    @Test
+    public void testGetForRedirectToLoginServlet() throws Exception {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        servlet.doGet(request, response);
+        validateRedirectedToLoginServlet(response);
+    }
+
+    @Test
+    public void testPostForRedirectToLoginServlet() throws Exception {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        servlet.doPost(request, response);
+        validateRedirectedToLoginServlet(response);
+    }
+
+    @Test
+    public void testPostGracefulRestart() throws Exception {
+        MockHttpServletRequest request = new MockHttpServletRequest();
+        request.getSession(true);
+        request.addParameter(GridForceRestartDelegateServlet.FORM_ID, "restart_nodes");
+        request.addParameter(GridForceRestartDelegateServlet.NODES, "127.0.0.1");
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Do not hard code the IP address](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2093723324/issues/source?bid=2928425&fileBranchId=3141724#l58)